### ISSUE TITLE
Snowflake ingestion errors for tables in schema with non-standard names

### ISSUE THREAD

**Issue by nickwu241** [linkedin/datahub#3531]
**Describe the bug**
Snowflake ingestion will fail if the ~~table~~ schema name contains non-standard characters e.g. dashes in their names.

Probably need to use double quotes to treat identifiers case-sensitively for these ~~tables~~ schemas. https://docs.snowflake.com/en/sql-reference/identifiers-syntax.html#double-quoted-identifiers

**To Reproduce**
1. Create a table with dashes in the ~~table~~ schema name
2. Ingest that table

sample ingestion recipe:
```
sink:
  type: console

source:
  type: snowflake
  config:
    # Coordinates
    host_port: <host_port>
    database_pattern:
      allow:
      - "database"
    schema_pattern:
      allow:
      - "schema-with-a-dash"
    table_pattern:
      allow:
      - "table"

    # Credentials
    username: <user>
    password: <password>
```

**Expected behavior**
Snowflake ingestion should succeed for db/schema/table names with non-standard characters.

**Screenshots**
NA

**Desktop (please complete the following information):**
NA

**Additional context**
```
File "/home/.venv/lib/python3.9/site-packages/datahub/entrypoints.py", line 91, in main
    sys.exit(datahub(standalone_mode=False, **kwargs))
File "/home/.venv/lib/python3.9/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
File "/home/.venv/lib/python3.9/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
File "/home/.venv/lib/python3.9/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
File "/home/.venv/lib/python3.9/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
File "/home/.venv/lib/python3.9/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
File "/home/.venv/lib/python3.9/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
File "/home/.venv/lib/python3.9/site-packages/datahub/cli/ingest_cli.py", line 58, in run
    pipeline.run()
File "/home/.venv/lib/python3.9/site-packages/datahub/ingestion/run/pipeline.py", line 125, in run
    for wu in self.source.get_workunits():
File "/home/.venv/lib/python3.9/site-packages/datahub/ingestion/source/sql/snowflake.py", line 228, in get_workunits
    for wu in super().get_workunits():
File "/home/.venv/lib/python3.9/site-packages/datahub/ingestion/source/sql/sql_common.py", line 328, in get_workunits
    yield from self.loop_tables(inspector, schema, sql_config)
File "/home/.venv/lib/python3.9/site-packages/datahub/ingestion/source/sql/sql_common.py", line 403, in loop_tables
    table_info: dict = inspector.get_table_comment(table, schema)  # type: ignore
File "/home/.venv/lib/python3.9/site-packages/sqlalchemy/engine/reflection.py", line 558, in get_table_comment
    return self.dialect.get_table_comment(
File "/home/.venv/lib/python3.9/site-packages/snowflake/sqlalchemy/snowdialect.py", line 606, in get_table_comment
    result = self._get_table_comment(connection, table_name, schema)
File "/home/.venv/lib/python3.9/site-packages/snowflake/sqlalchemy/snowdialect.py", line 584, in _get_table_comment
    cursor = connection.execute(sql_command)
File "/home/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1003, in execute
    return self._execute_text(object_, multiparams, params)
File "/home/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1172, in _execute_text
    ret = self._execute_context(
File "/home/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1316, in _execute_context
    self._handle_dbapi_exception(
File "/home/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1510, in _handle_dbapi_exception
    util.raise_(
File "/home/.venv/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
File "/home/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1276, in _execute_context
    self.dialect.do_execute(
File "/home/.venv/lib/python3.9/site-packages/sqlalchemy/engine/default.py", line 608, in do_execute
    cursor.execute(statement, parameters)
File "/home/.venv/lib/python3.9/site-packages/snowflake/connector/cursor.py", line 781, in execute
    Error.errorhandler_wrapper(
File "/home/.venv/lib/python3.9/site-packages/snowflake/connector/errors.py", line 268, in errorhandler_wrapper
    handed_over = Error.hand_to_other_handler(
File "/home/.venv/lib/python3.9/site-packages/snowflake/connector/errors.py", line 323, in hand_to_other_handler
    cursor.errorhandler(connection, cursor, error_class, error_value)
File "/home/.venv/lib/python3.9/site-packages/snowflake/connector/errors.py", line 202, in default_errorhandler
    raise error_class(

ProgrammingError: (snowflake.connector.errors.ProgrammingError) 001003 (42000): SQL compilation error:
syntax error line 1 at position 84 unexpected '-'.
[SQL: SHOW /* sqlalchemy:_get_table_comment */ TABLES LIKE 'redacted_table_name' IN SCHEMA schema-with-a-dash]
(Background on this error at: http://sqlalche.me/e/13/f405)
```

**Comment by varunbharill** [comment]
Hi @nickwu241 ,
I tried reproducing the issue and created a test table in snowflake with a "-" in it. Upon running the ingestion, I can ingest it fine. (Although, I do need to try with other symbols especially ".", which I suspect might fail but due to other reasons). But overall it seems fine. Could you please share a stacktrace?

<img width="309" alt="Screen Shot 2021-11-10 at 11 33 17 AM" src="https://user-images.githubusercontent.com/2242683/141180926-44bcd4db-db4a-4e7a-b4ec-3c45d1e01989.png">

**Comment by nickwu241** [comment]
@varunbharill Thanks for trying to re-pro it, sorry the re-pro is actually a SCHEMA with a dash in the name, and then the tables are okay (rather than what I originally wrote which was a table with a dash)

e.g. `demo_db.public-with-dash.table` will probably fail -- i included a stacktrace in the original issue now.

**Comment by shirshanka** [comment]
Fixed via #3635 and released via `acryl-datahub==0.8.17.3`
Please re-open if this is still an issue!

**Comment by nickwu241** [comment]
Awesome! Thanks!
